lib/metadata/metadataUtils.js

```diff
@@ -238,6 +238,16 @@ function standardMetadataValidateBucketAndObj(params, actionImplicitDenies, log,
             }
             return next(null, bucket, objMD);
         },
+        (bucket, objMD, next) => {
+            if (params.disableQuota || (!params.enableQuota && !requestType.some(type => actionNeedQuotaCheck[type] ||
+                actionWithDataDeletion[type]))) {
```

Comment:
nit: API may be simpler with a single parameter (`checkQuota` ?) instead of two parameters (`disableQuota` and `enableQuota`)
* `checkQuota === undefined` : automatically depending on the action
* `checkQuota === true` : always check quota
* `checkQuota === false` : never check quota

```suggestion
            const needQuotaCheck = requestType -> requestType.some(type => actionNeedQuotaCheck[type] || actionWithDataDeletion[type]);
            const checkQuota = params.checkQuota === undefined ? needQuotaCheck(requestType) : params.checkQuota;
            if (!checkQuota) {
```

Reply:
Thanks, I cannot use your suggestion directly as I need to update the logic, but this works well indeed - PR was updated